tableau-databricks/connection-builder.js

```diff
@@ -16,20 +16,35 @@ limitations under the License.
 */
 
 (function dsbuilder(attr) {
+	var vendorDefined = {
+		attributeHttpPath: "v-http-path"
+	}
+
 	var params = {};
 
 	// The Databricks cluster ODBC endpoint
-	params["HOST"] = attr["server"];
+	params["HOST"] = attr[connectionHelper.attributeServer];
 	params["PORT"] = "443";
-	params["HTTPPATH"] = attr["dbname"];
+	params["HTTPPATH"] = attr[vendorDefined.attributeHttpPath];
 	params["THRIFTTRANSPORT"] = "2";
 	params["SPARKSERVERTYPE"] = "3";
 	params["SSL"] = "1";
 
-	// Authentication by username and password only
-	params["AUTHMECH"] = 3;
-	params["UID"] = attr["username"];
-	params["PWD"] = attr["password"];
+	var authenticationMode = attr[connectionHelper.attributeAuthentication];
```

Comment:
is this detecting if you're passing a token or password?

Reply:
Yes, `attributeAuthentication` is the authentication mode.